community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -696,22 +694,23 @@ private static TreeState other( Pair<TreeState,TreeState> states, TreeState stat
      *
      * @param pagedFile {@link PagedFile} to read the state pages from.
      * @return both read state pages.
-     * @throws IOException if state pages are missing (file is smaller than that) or if they are both empty.
+     * @throws MetadataMismatchException if state pages are missing (file is smaller than that) or if they are both empty.
+     * @throws IOException on {@link PageCursor} error.
      */
-    private static Pair<TreeState,TreeState> loadStatePages( PagedFile pagedFile ) throws IOException
+    private static Pair<TreeState,TreeState> loadStatePages( PagedFile pagedFile ) throws MetadataMismatchException, IOException
     {
         try
         {
             Pair<TreeState,TreeState> states = readStatePages( pagedFile );
             if ( states.getLeft().isEmpty() && states.getRight().isEmpty() )
             {
-                throw new IOException( "Index is not fully initialized since its state pages are empty" );
+                throw new MetadataMismatchException( "Index is not fully initialized since its state pages are empty" );
```

Comment:
to be honest, i do not really see why exceptions that are thrown here become MetadataMismatchException, but can be that its my fresh out of the story view

Reply:
Unifying behaviour between missing state page and meta page. Both of them means file was not fully initialised. And also we want to differentiate this case from an IOException caused by PageCursor not being able to read / write.